Packages/com.unity.inputsystem/Documentation~/Interactions.md

```diff
@@ -69,11 +73,30 @@ If you have multiple Controls bound to a Binding or an Action which has an Inter
 
+At any one time, only one Interaction can be "driving" the action, i.e. gets to determine the action's current [`phase`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_phase). If an Interaction higher up in the stack cancels, Interactions lower down in the stack get a chance at taking over.
+
+### Timeouts
+
+Interactions may require to wait a certain time for a specific input to occur or to not occur. An example of this is the [Hold](#hold) interaction which, after a button is pressed, has to wait for a set [duration](../api/UnityEngine.InputSystem.Interactions.HoldInteraction.html#UnityEngine_InputSystem_Interactions_HoldInteraction_duration) until the "hold" is complete. To do this, an interaction will install a timeout using [`SetTimeout`](../api/UnityEngine.InputSystem.InputInteractionContext.html#UnityEngine_InputSystem_InputInteractionContext_SetTimeout_System_Single_).
+
+It can be useful to know how much of a timeout is left for an interaction to complete. For example, you may want to display a bar in the UI that is charging up while the interaction is waiting to complete. You can query the percentage to which a timeout has completed using [`GetTimeoutCompletionPercentage`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_GetTimeoutCompletionPercentage_).
+
+```CSharp
+// Returns a value between 0 (inclusive) and 1 (inclusive).
+var warpActionCompletion = playerInput.actions["warp"].GetTimeoutCompletionPercentage();
+```
+
+Note that each Interaction can have its own separate timeout (but only a single one at any one time). If [multiple interactions](#multiple-interactions-on-a-binding) are in effect, then [`GetTimeoutCompletionPercentage`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_GetTimeoutCompletionPercentage_) will use the timeout of only the one interaction that is currently driving the action.
+
+Some Interactions may involve multiple timeouts in succession. In this case, knowing only the completion of the currently running timeout (if any) is often not useful. An example is [`MultiTapInteraction`](../api/UnityEngine.InputSystem.Interactions.MultiTapInteraction.html) which involves a timeout on each individual tap as well as a timeout in-between taps. The Interaction is complete only after a full tap sequence has been performed.
+
+An Interaction can inform the Input System of the total time it will run timeouts for using [`SetTotalTimeoutCompletionTime`](../api/UnityEngine.InputSystem.InputInteractionContext.html#UnityEngine_InputSystem_InputInteractionContext_SetTotalTimeoutCompletionTime_System_Single_).
```

Comment:
```suggestion
An Interaction can use [`SetTotalTimeoutCompletionTime`](../api/UnityEngine.InputSystem.InputInteractionContext.html#UnityEngine_InputSystem_InputInteractionContext_SetTotalTimeoutCompletionTime_System_Single_) to inform the Input System of the total time it will run timeouts for.
```

Reply:
Pushed